prefetch.bs

```diff
@@ -637,23 +624,14 @@ The <dfn>list of sufficiently strict speculative navigation referrer policies</d
         :  [=session history entry/document state=]
         :: |documentState|
     1. Let |request| be the result of [=creating a navigation request=] given |entry|, |document|'s [=relevant settings object=], |document|'s [=node navigable=]'s [=navigable/container=], and false.
-    1. If |prefetchRecord|'s [=prefetch record/partition state=] is a [=cross-partition prefetch state=]:
-        1. Clear |request|'s [=request/use-URL-credentials flag=].
-        1. Set |request|'s [=request/cache mode=] to "`no-store`".
-
-            <div class="issue">This ends up setting the `` `Cache-Control` `` and `` `Pragma` `` request headers, which is contrary to what Chromium does today when skipping cache here. One approach would be to add a flag similar to [=request/prevent no-cache cache-control header modification flag=]. It would also be possible to have a cache that can be copied into the ordinary cache, like Chromium does for cookies.</div>
-
-        1. If |documentState|'s [=document state/request referrer policy=] is not in the [=list of sufficiently strict speculative navigation referrer policies=], then [=prefetch record/cancel and discard=] |prefetchRecord| given |document|, and return.
-
-        <div class="note">Though credentials are still included, they will be isolated such that no credentials are present to begin with.</div>
     1. Let |coopEnforcementResult| be the result of [=creating a cross-origin opener policy enforcement result for navigation=] given |document| and |document|'s [=Document/origin=].
     1. Let |global| be |document|'s [=relevant global object=].
     1. [=In parallel=]:
         1. Let |navigationParams| be the result of [=creating navigation params by fetching=] given |request|, |entry|, |coopEnforcementResult|, |document|'s [=node navigable=], |sourceSnapshotParams|, |targetSnapshotParams|, "`other`", null (navigationId), "`navigate`", and <a href="#create-navigation-params-by-fetching-prefetchRecord"><i>prefetchRecord</i></a> |prefetchRecord|.
         1. If |navigationParams|'s [=navigation params/response=] does not [=support prefetch=], then set |navigationParams| to null.
-        1. If |prefetchRecord|'s [=prefetch record/partition state=] is a [=cross-partition prefetch state=] whose [=cross-partition prefetch state/origins with conflicting credentials=] is not [=set/empty=], then set |navigationParams| to null.
+        1. If |prefetchRecord|'s [=prefetch record/origins with conflicting credentials=] is not [=set/empty=], then set |navigationParams| to null.
 
```

Comment:
Does this mean we still follow the redirects, even after we know the end result will be failure? Is that a good idea?

It seems like instead of tracking the list of origins with conflicting credentials, which we only ever do a size check on (right?), we could just bail out with a network error instead of appending to the list.

Reply:
It could be a boolean; it's been a list just because it's about as long and seemed to me a little clearer about what's going on, and UAs might want to report it anyway. But that might be overkill, I'm not attached to that.

As far as bailing out with a network error, the problem is that it would reveal whether the user has credentials to an origin that shouldn't know. Consider:

a.example wants to know if the user has credentials on b.example, and there is a URL on b.example which redirects to a.example (unconditionally).

Then a page on a.example can request a prefetch of https://b.example/redirect-to/a and observe whether or not it sees a request for a.example. This should not be possible, because we're not in the b.example partition, until such time as the user actually navigates.

The private prefetching proxy also sends these requests that we know (at least, before Supports-Loading-Mode: uncredentialed-prefetch) will not be used, so that this information is not leaked at prefetch time.

Reply:
This makes perfect sense. We should document that in a note, probably at the point at which we add to the set, but maybe with another small note here.

I think a boolean is probably a bit cleaner, personally. Otherwise I as the reader am wondering if I missed something. Or I guess we could add some text to make use of the origins, like "User agents may [=report a warning to the console=] describing how |prefetchRecord|'s [=prefetch record/origins with conflicting credentials=] prevented the prefetch from being used."

Reply:
Switched to a boolean.